packages/admin/cms-admin/src/blocks/createRichTextBlock.tsx

```diff
@@ -179,7 +188,55 @@ export const createRichTextBlock = (
                 <SelectPreviewComponent>
                     <Rte
                         minHeight={options.minHeight}
-                        options={rteOptions}
+                        options={{
+                            ...rteOptions,
+                            draftJsProps: {
+                                // Naive implementation of https://github.com/facebook/draft-js/blob/main/src/component/handlers/edit/editOnPaste.js
+                                // FIXME
+                                // @ts-expect-error handlePastedText is not exposed to draftJsProps in RTE
+                                handlePastedText: (text, html) => {
+                                    if (html) {
+                                        const { contentBlocks } = convertFromHTML(html);
+
+                                        if (contentBlocks) {
+                                            const htmlMap = BlockMapBuilder.createFromArray(contentBlocks);
+
+                                            let newContent = Modifier.replaceWithFragment(
+                                                state.editorState.getCurrentContent(),
+                                                state.editorState.getSelection(),
+                                                htmlMap,
+                                            );
+
+                                            const entries = (newContent.getEntityMap().__getAll() as Immutable.Map<string, EntityInstance>).entries();
+
+                                            // @ts-expect-error Immutable.Map#entries is iterable, but missing [Symbol.iterator]()
+                                            for (const [key, entity] of entries) {
+                                                if (entity.getType() === "LINK") {
+                                                    const data = entity.getData();
+
+                                                    if (data.url) {
```

Comment:
where does this come from? is this the default behaviour?

Reply:
Yes, I do believe so. I've determined this by reviewing the `entity.getData()` response.

In addition, the Draft.js documentation also uses `url` as example for the link entity:

<img width="828" alt="Screen Shot 2022-08-02 at 14 57 14" src="https://user-images.githubusercontent.com/48853629/182380438-cc93b816-83a2-44a9-9da6-141339d0f9a7.png">

https://draftjs.org/docs/advanced-topics-entities/#creating-and-retrieving-entities